### PR TITLE
Draft of changes needed to use distill in wasm

### PR DESCRIPTION
**Pull request by aclysma**
Will clean this up and rebase, this draft PR is just to make it easy to show the changes necessary.

### CODE REVIEW

**Comment by kabergstrom** [comment]
Looks great!!

**Comment by DavidVonDerau** [comment]
@aclysma When you rebase you can get rid of the crate version bump since @kabergstrom pushed that the other day.

File: loader/src/packfile_io.rs [hunk 1]
```diff
@@ -140,11 +141,13 @@ impl PackfileReader {
                 .or_insert_with(|| vec![idx as u32]);
         }
 
+        let runtime = tokio::runtime::Builder::new_current_thread().build()?;
```
Comment (DavidVonDerau) [inline loader/src/packfile_io.rs:148]: Is it possible to pass a `use single threaded` flag somehow? e.g. when configuring loaders or `distill`

File: loader/src/loader.rs [hunk 2]
```diff
@@ -83,7 +84,7 @@ struct AssetVersionLoad {
 #[derive(Debug)]
 struct AssetLoad {
     asset_id: AssetUuid,
-    last_state_change_instant: std::time::Instant,
```
Comment (DavidVonDerau) [inline loader/src/loader.rs:86]: If `std::time` is banned, do you think I'll have any issues with my work on removing `tokio`? I'm wondering if any of the dependencies I'm using are expecting `std::time`.

File: loader/src/packfile_io.rs [hunk 3]
```diff
@@ -52,20 +66,69 @@ impl Drop for PackfileMessageReader {
...
+            data,
+            message_reader: ManuallyDrop::new(ThreadLocal::new()),
+        })
+    }
+}
+
+impl PackfileMessageReader for PackfileMessageReaderBuffer {
+    fn get_reader(&self) -> capnp::Result<pack_file::Reader<'_>> {
+        let message_reader = self.message_reader.get_or_try(|| {
+            // We ensure that the reader is dropped before the mmap so it's ok to cast to 'static here
```
Comment (DavidVonDerau) [inline loader/src/packfile_io.rs:87]: Commented out code?

Comment (DavidVonDerau) [reply loader/src/packfile_io.rs:87]: Looks like this block happens above at line 50 now.

File: loader/src/packfile_io.rs [hunk 4]
```diff
@@ -8,42 +8,56 @@ use std::{
...
+}
 
+#[cfg(not(target_arch = "wasm32"))]
+impl PackfileMessageReader for PackfileMessageReaderFile {
     fn get_reader(&self) -> capnp::Result<pack_file::Reader<'_>> {
-        let messge_reader = self.message_reader.get_or_try(|| {
+        let message_reader = self.message_reader.get_or_try(|| {
             // We ensure that the reader is dropped before the mmap so it's ok to cast to 'static here
             let slice: &[u8] = &self.mmap;
             let mut slice: &[u8] = unsafe { std::mem::transmute::<&[u8], &'static [u8]>(slice) };
```
Comment (DavidVonDerau) [inline loader/src/packfile_io.rs:50]: Should probably put the comment into the `unsafe` block and prefix it with `SAFETY: `.

File: loader/src/loader.rs (hunk 2 above)
Comment (aclysma) [reply loader/src/loader.rs:86]: Honestly not sure, I think we'd have to try it and see. Won't know until it fails at runtime unfortunately.

File: loader/src/packfile_io.rs (hunk 3 above)
Comment (aclysma) [reply loader/src/packfile_io.rs:87]: This is cleaned up in the other PR

**Comment by aclysma** [comment]
Replaced by #98

**Pull request closed without merging**